src/Sylius/Behat/Page/Shop/Product/IndexPage.php

```diff
@@ -24,31 +24,31 @@ public function getRouteName(): string
 
     public function countProductsItems(): int
     {
-        $productsList = $this->getDocument()->find('css', '#products');
+        $productsList = $this->getDocument()->find('css', '[data-test-products]');
 
-        $products = $productsList->findAll('css', '.card');
+        $products = $productsList->findAll('css', '[data-test-product]');
 
         return count($products);
     }
 
     public function getFirstProductNameFromList(): string
     {
-        $productsList = $this->getDocument()->find('css', '#products');
+        $productsList = $this->getDocument()->find('css', '[data-test-products]');
 
-        return $productsList->find('css', '.card:first-child .content > a')->getText();
+        return $productsList->find('css', '[data-test-product]:first-child [data-test-product-content] > a')->getText();
```

Comment:
IMO you should add attribtute to this link  instead use '> a'

Reply:
If i good remember we talking that good practice will be use data-attributes and if we need html attributes, we go to working without css class.